File: app/src/main/java/net/mmho/photomap2/SearchResultDialogFragment.java
package net.mmho.photomap2;

import android.app.AlertDialog;
import android.app.Dialog;
import android.location.Address;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.DialogFragment;
import android.support.v4.app.Fragment;

import com.google.android.gms.maps.CameraUpdate;
import com.google.android.gms.maps.CameraUpdateFactory;

import java.util.List;

public class SearchResultDialogFragment extends DialogFragment{
    static SearchResultDialogFragment newInstance(String location, List<Address> addresses){
        SearchResultDialogFragment fragment = new SearchResultDialogFragment();


        Bundle bundle = new Bundle();
        bundle.putString("title",location);
        bundle.putParcelableArray("address", addresses.toArray(new Address[addresses.size()]));
        fragment.setArguments(bundle);
        return fragment;
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        String title = getArguments().getString("title");
        final Address list[] = (Address[]) getArguments().getParcelableArray("address");
        AddressListAdapter adapter = new AddressListAdapter(getActivity(),android.R.layout.simple_list_item_2,list);
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
        builder.setTitle(getString(R.string.search_title, title));
        builder.setAdapter(adapter, (dialog, which) -> {
            Fragment f = getActivity().getSupportFragmentManager().findFragmentById(R.id.map);
            if(f instanceof PhotoMapFragment){
                final CameraUpdate update = CameraUpdateFactory.newLatLngZoom(AddressUtil.INSTANCE.addressToLatLng(list[which]),
                    PhotoMapFragment.Companion.getDEFAULT_ZOOM());
                ((PhotoMapFragment)f).getMapAsync(map->map.moveCamera(update));
            }
        });
        return builder.create();

    }
}

File: app/src/main/kotlin/net/mmho/photomap2/SearchResultDialogFragment.kt
package net.mmho.photomap2

import android.R.layout.simple_list_item_2
import android.app.AlertDialog
import android.app.Dialog
import android.location.Address
import android.os.Bundle
import android.support.v4.app.DialogFragment
import com.google.android.gms.maps.CameraUpdateFactory

class SearchResultDialogFragment : DialogFragment() {

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val title = arguments?.getString("title")
        @Suppress("UNCHECKED_CAST")
        val list = arguments?.getParcelableArray("address") as Array<Address>
        val adapter = AddressListAdapter(requireActivity(),simple_list_item_2, list)
        val builder = AlertDialog.Builder(activity)
        builder.setTitle(getString(net.mmho.photomap2.R.string.search_title, title))
        builder.setAdapter(adapter) { _, which ->
            val f = requireActivity().supportFragmentManager.findFragmentById(net.mmho.photomap2.R.id.map)
            if (f is PhotoMapFragment) {
                val update = CameraUpdateFactory.newLatLngZoom(list[which].toLatLng(),
                    PhotoMapFragment.DEFAULT_ZOOM)
                f.getMapAsync { map -> map.moveCamera(update) }
            }
        }
        return builder.create()

    }

    companion object {
        internal fun newInstance(location: String, addresses: List<Address>): SearchResultDialogFragment {
            val fragment = SearchResultDialogFragment()


            val bundle = Bundle()
            bundle.putString("title", location)
            bundle.putParcelableArray("address", addresses.toTypedArray())
            fragment.arguments = bundle
            return fragment
        }
    }
}
